features/affordability.feature

```diff
@@ -78,8 +78,97 @@ Scenario: Recalculate details for two applicants
+  | 40000  | 0     | 2512.89 | 35000   | 0      | 2229.56  | 300    | 0     | 0    | 600    | 325   | 0    | 400  | 0    | 300  | 25   | 3    | £199,920 | £299,880 | 249,900   | £1,185.05 | £1,632.40 | £1,225.00 | 51%      | £2,410.05 | 49%      | £2,332.40 | £1,610.11 | £1,207.34 | £0.00    |
+  | 100000 | 10000 | 6000    | 50000   | 5000   | 3000     | 300    | 300   | 300  | 300    | 300   | 300  | 300  | 300  | 300  | 25   | 3    | £441,840 | £662,760 | 552,300   | £2,619.07 | £3,980.93 | £1,500.00 | 46%      | £4,119.07 | 54%      | £4,880.93 | £3,558.48 | £3,041.52 | £300.00  |
+  | 25000  | 0     | 1500    | 33000   | 0      | 1750     | 750    | 0     | 0    | 0      | 0     | 0    | 0    | 0    | 0    | 25   | 4    | £137,200 | £205,800 | 171,500   | £905.24   | £1,594.76 | £750.00   | 51%      | £1,655.24 | 49%      | £1,594.76 | £1,212.13 | £1,287.87 | £0.00    |
+  | 25000  | 0     | 1500    | 33000   | 0      | 1750     | 750    | 0     | 0    | 0      | 0     | 0    | 0    | 0    | 0    | 33   | 2    | £137,200 | £205,800 | 171,500   | £591.95   | £1,908.05 | £750.00   | 41%      | £1,341.95 | 59%      | £1,908.05 | £885.16   | £1,614.84 | £0.00    |
+  | 50000  | 0     | 2714.22 | 50000   | 0      | 2714.22  | 0      | 0     | 0    | 0      | 0     | 420  | 0    | 0    | 0    | 25   | 3    | £280,000 | £420,000 | 350,000   | £1,659.74 | £3,768.70 | £0.00     | 31%      | £1,659.74 | 69%      | £3,768.70 | £2,255.05 | £3,173.39 | £420.00  |
+
+Scenario: Repayment term tooltip
+  Given I visit the affordability page
+  When  I enter all details for single applicant
+  Then  I should see the repayment term tooltip
+
+@javascript
+Scenario: Adjusting the interest rate in JavaScript
+  Given I visit the affordability page
+  When  I enter all details for single applicant
+  Then  I should see the correct results for a single applicant for JavaScript
+  And   I should be able to tweak the results by adjusting the interest rate text for JavaScript
+
+@javascript
+Scenario: Adjusting the interest rate slider
+  Given I visit the affordability page
+  When  I enter all details for single applicant
+  Then  I should see the correct results for a single applicant for JavaScript
+  And   I should be able to tweak the results by adjusting the interest rate slider
+
+@javascript
+Scenario: Adjusting the mortgage term in JavaScript
+  Given I visit the affordability page
+  When  I enter all details for single applicant
+  Then  I should see the correct results for a single applicant for JavaScript
+  And   I should be able to tweak the results by adjusting the mortgage term text for JavaScript
+
+@javascript
+Scenario: Adjusting the mortgage term slider
+  Given I visit the affordability page
+  When  I enter all details for single applicant
+  Then  I should see the correct results for a single applicant for JavaScript
```

Comment:
Indeed, it looks like the step is differentiating between itself and another step which does the same check but for non JS tests.

In that case it might be nice to have a single step that's aware of what kind of test it is and behaves accordingly.  I've done this kind of thing some time in the past but don't have anything examples to hand, but there's some cleverness [in the frontend repo](https://github.com/moneyadviceservice/frontend/blob/master/features/support/cucumber_extensions.rb) which might provide some inspiration.